src/Native/System.Security.Cryptography.Native/pal_eckey.cpp

```diff
@@ -12,33 +12,59 @@ extern "C" void CryptoNative_EcKeyDestroy(EC_KEY* r)
     EC_KEY_free(r);
 }
 
-extern "C" EC_KEY* CryptoNative_EcKeyCreateByCurveName(int32_t nid)
+extern "C" EC_KEY* CryptoNative_EcKeyCreateByOid(const char* oid)
 {
+    // oid can be friendly name or value
+    int nid = OBJ_txt2nid(oid);
     return EC_KEY_new_by_curve_name(nid);
 }
 
 extern "C" int32_t CryptoNative_EcKeyGenerateKey(EC_KEY* eckey)
 {
-    return EC_KEY_generate_key(eckey);
+    if (!EC_KEY_generate_key(eckey))
+        return 0;
+
+    return EC_KEY_check_key(eckey);
 }
 
 extern "C" int32_t CryptoNative_EcKeyUpRef(EC_KEY* r)
 {
     return EC_KEY_up_ref(r);
 }
 
-extern "C" int32_t CryptoNative_EcKeyGetCurveName(const EC_KEY* key)
+extern "C" int32_t CryptoNative_EcKeyGetSize(const EC_KEY* key, int32_t* keySize)
 {
-    if (key == nullptr)
-    {
-        return NID_undef;
-    }
+    if (!keySize)
+        return 0;
+    
+    *keySize = 0;
+
+    if (!key)
+        return 0;
+
+    const EC_GROUP* group = EC_KEY_get0_group(key);
+    if (!group)
+        return 0;
+
+    *keySize = EC_GROUP_get_degree(group);
+
+    return 1;
+}
+
+extern "C" int32_t CryptoNative_EcKeyGetOidValueCurveName(const EC_KEY* key, int32_t* nidName)
```

Comment:
I'm not sure that "OidValueCurveName" makes a lot of sense, particularly since it's emitting a nid, not a name.